### DISCUSSION TITLE
[FIXED PR#303] Interface does not translate - HA language not detected

### DISCUSSION THREAD

**Discussion by adnovea** [slipx06/sunsynk-power-flow-card#226] [General]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I have installed all the update for a couple of months and though the language translation issue will be solved. 
Now I guess it's time to report this issue.
My HA interface is in French but the add-on labels are in English.
Looking at the SRC folder, there is a file for french translation and inside the sunsynk-power-flow-card.js file I can found the french strings.
I have no clue what could be wrong.

### Expected behaviour

Card strings should translate

### Card Version

2.10.0

### Home Assistant Version

2023.10.0

### Configuration

```shell
type: custom:sunsynk-power-flow-card
cardstyle: full
panel_mode: false
large_font: false
show_solar: true
show_battery: true
inverter:
  modern: true
  colour: lightgrey
  autarky: power
  model: sunsynk
  auto_scale: true
battery:
  energy: 0
  shutdown_soc: 20
  invert_power: true
  colour: '#f06292'
  show_daily: true
  animation_speed: 6
  max_power: 2000
  full_capacity: 100
  empty_capacity: 0
  show_absolute: true
  auto_scale: false
solar:
  colour: orange
  show_daily: true
  mppts: 2
  animation_speed: 9
  max_power: 4920
  pv1_name: Ouest
  pv2_name: Est
  auto_scale: false
load:
  colour: '#5fb6ad'
  show_daily: true
  show_daily_aux: true
  show_aux: true
  invert_aux: false
  animation_speed: 8
  max_power: 5000
  aux_name: Surplus (V.E.)
  show_absolute_aux: true
  aux_type: default
  aux_colour: lightblue
  aux_off_colour: lightgrey
  aux_loads: 0
  aux_load1_name: Véhicule
  aux_load1_icon: mdi:car-electric-outline
  aux_load2_name: Piscine
  aux_load2_icon: mdi:pool
  additional_loads: 0
  load1_name: Secouru
  load1_icon: mdi:power-plug-battery-outline
  auto_scale: false
grid:
  colour: '#488fc2'
  no_grid_colour: rgb(192,0,0)
  show_daily_buy: true
  show_daily_sell: true
  show_nonessential: true
  nonessential_icon: default
  nonessential_name: Gros consommateurs
  additional_loads: 0
  load1_name: Non essentielles
  load1_icon: mdi:washing-machine
  invert_grid: false
  animation_speed: 8
  max_power: 6000
  auto_scale: false
entities:
```

**Comment by slipx06** [comment]
Hi. This does work if you explicitly set you HA Dashboard language
![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/7227275/49e8836c-a59c-4d1f-8603-c90184913b26)

The problem is the translation does not look good and needs some work to fix alignment

![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/7227275/3bbc3516-9d0d-4b6e-9bf0-c2e2818f0c81)

**Comment by adnovea** [comment]
Hi splix06,

Thanks for your quick reply.
I had the language set under Parametres > System > General

![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/44359861/136e84a5-1275-4167-bee9-c0ce43ea2470)

but the add-on always displays English strings

![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/44359861/c82c0168-9b33-444d-83fa-fadefa9bb0b8)

Any idea what could be wrong ?

French is wordy so we need to find some shortcuts.
It could be something like :
```
    "common": {
        "description": "Une carte Home Assistant pour émuler les flux d'énergie affichés à l'écran des onduleurs Sunsynk ",
        "version": "Version",
        "runtime_to": "AVANT D'ATTEINDRE",
        "battery_floating": "BATTERIE FLOTTANTE",
        "charge": "CHARGE",
        "to": "À",
        "daily_load": "CHARGE DU JOUR",
        "daily_aux": "CHARGE AUX DU JOUR",
        "daily_grid_buy": "ACHAT CE JOUR",
        "daily_grid_sell": "VENTE CE JOUR",
        "daily_solar": "PRODUCTION DU JOUR",
        "daily_solar_left": "PRODUCTION / RESTANT (JOUR)",
        "daily_charge": "CHARGE DU JOUR",
        "daily_discharge": "DÉCHARGE DU JOUR",
        "ratio": "Ratio",
        "timer_off": "Prog. M.",
        "timer_on": "Prog. A.",
        "priority_batt": "Batterie en 1er",
        "priority_load": "Charges en 1er",
        "nonessential_name": "Non Essentielles",
        "aux_name": "Auxiliaire",
        "load1_name": "",
        "load2_name": "",
        "pv1_name": "PV1",
        "pv2_name": "PV2",
        "pv3_name": "PV3",
        "pv4_name": "PV4",
        "essential": "Essentielles"
    },

```

**Comment by slipx06** [comment]
Thanks I will update with your suggestions.
You need to set the language under your profile for the card translation to work.

**Comment by adnovea** [comment]
Thanks for the hint but it was already the case.
![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/44359861/c9c3555a-d660-45e8-b15e-f5f4c1d91f4b)

So what I did was to select Deutsche and the translation worked,.
I selected back French and it worked.
I logout and login again but the card is back in English. 
Look like you must re-select language each time you login.

**Comment by slipx06** [comment]
Thanks for confirming the solution. That might be a HA bug or maybe your browser is caching some files. I'm not sure. You can try clear your browser cache or double check in a new incognito window........or just don't log out :)

**Comment by adnovea** [comment]
Hi, I already did the test using an other browser (Edge instead of FF)  and this is how I found the strange behavior.
Clearing the browser cache reset the language selection. You must stay logged after changing the user's language.
Many thanks for you nice add-on. I run it for a DEYE 5K.

**Comment by adnovea** [comment]
Oups, timer_off/on inverted in my French translation.
I suggest the following to solve this issue. THX
```
        "timer_off": "Prog. Off.",
        "timer_on": "Prog. On",
```
Where "Autarky" can be translated to "Autoconso" ?
Could decimals of "energy_cost_buy" be selected 0-2 ?

**Comment by slipx06** [comment]
Thanks I will fix that. I want to leave the term Autarky as it is a concept of self sufficiency that could get lost in translation. I will add an additional attribute to the Grid section, `energy_cost_decimals: 2` so you can customise the number of decimal places for energy cost buy and sell

**Comment by slipx06** [comment]
> Oups, timer_off/on inverted in my French translation. I suggest the following to solve this issue. THX
> 
> ```
>         "timer_off": "Prog. Off.",
>         "timer_on": "Prog. On",
> ```
> 
> Where "Autarky" can be translated to "Autoconso" ? Could decimals of "energy_cost_buy" be selected 0-2 ?

@adnovea  Fixed and added in v2.12.0

**Comment by adnovea** [comment]
Thanks for you quick support and for your work.

**Comment by tigers75** [comment]
I think this should be added to the main page FAQ, it's not the expected behaviour and it's very rare that people track down a closed issue on GitHub

**Comment by slipx06** [comment]
When you log a bug report I do ask that you please check open and closed issues

![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/7227275/824def97-2da0-4b09-8837-a61513ed106f)

If more people trip over this I'll add it to the main page.

**Comment by tigers75** [comment]
Well, I am aware and I did search, but couldn't find it. So will many that will come after me.
Also since this is the only add-in that doesn't get translated properly I believe the issue is NOT resolved, and should be investigated more and not closed.
Maybe it happens only to who updated the card, and not on fresh installs. Anyway I don't think it's normal.

**Comment by slipx06** [comment]
@tigers75 I've made some updates to the translation logic.  Can you test [3.1.0](https://github.com/slipx06/sunsynk-power-flow-card/releases/tag/v3.1.0) if it solves the issue

**Comment by tux-84** [comment]
I also encountered this issue and I am staying logged in. In Fully Kiosk Browser on my Huawei Tab, the language in my profile is set to German but the cad shows English, while on my Desktop, the card shows German. All other cards on my Tablet show up in German on both devices, so this is clearly a card issue. @slipx06 please re-open this issue.

PS: Bought you 3 coffees anyway, since this card is very nice and offers so much functionality.

**Comment by slipx06** [comment]
@jzielke84 can you try changing your language to English and then back to German on your tablet.

**Comment by tux-84** [comment]
@slipx06 This worked, but wth 😳  Why did all other cards show up correctly? Doing this would have never occoured to me. Thank you!

**Comment by ollelinse** [comment]
Ein wirklich nützliches und gut gemachtes Tool, Danke für die Bereitstellung. Aber auch ich bekomme die Zusatzbezeichnungen nicht auf Deutsch eingestellt. Es bleibt Englisch, egal was ich mache. Ich habe alle Hinweise getestet, die hier vorgeschlagen wurden.

**Reply by ollelinse** [reply]
Das Problem hat sich erledigt. Lösung:
Man darf die Sprache **nicht in den Systemeinstellungen** von Home Assistant ändern, sondern muss dies im Benutzerprofil auf englisch und wieder zurück auf die gewünschte Sprache stellen. Wird mit mehreren Benutzern gearbeitet, muss dies auf jeder Benutzerebene  getrennt erfolgen.
Vielleicht lässt sich für neue Anwender dieses schönen Tools ja noch etwas optimieren, was dieses kleine Ärgernis umgeht.

**Comment by slipx06** [comment]
Thanks. Yes you need to set this under the user profile. I think the issue has to do with the way the card is using SVG elements and the how the browser caches them.  At least there is a known workaround

**Comment by Pho3niX90** [comment]
This should be fixed in PR: #303

**Comment by Pho3niX90** [comment]
@adnovea @tigers75 @jzielke84 could you confirm this is now working for you?

**Comment by adnovea** [comment]
**[FIXED]**
Now it works fine at login

THX for constantly improving this very useful add-on

_Previous behavior :
Did not work until you changed the language back and forth.
Then the language remained correct until you logged out.
At login, whatever your selected language was, the card displayed English texts._

**Reply by Pho3niX90** [reply]
Perfect. Thanks for the feedback, just mark it as resolved
